CONTRIBUTING.md

```diff
@@ -0,0 +1,134 @@
+# CONTRIBUTING
+
+Thanks for contributing!
+We welcome any and all kinds of contribution, including:
+- bug fixes and code contributions
+- ideas for new features
+- testing
+- documentation improvements
+- general suggestions on how we can do better
+- ...
+
+
+## Development environment setup
+
+This project is using [hatch](https://hatch.pypa.io), a modern Python project manager.
+Please refer to their [official documentation](https://hatch.pypa.io/latest/install/) for installation instructions for your system.
+
+After installation, you can use ``hatch`` to execute specific tasks, which are defined in ``pyproject.toml`` (see individual subheadings below).
+Alternatively, ``hatch`` can provide you with a development shell with the necessary packages:
+
+```
+hatch shell
+```
+
+Presently, this work require a patched linkml installation.
+The patch(es) are listed in `patches/`.
+The script `tools/patch_linkml` documents how they need to be applied.
+This script can be used to patch a local installation, and is also executed in the actions that validate the included data models.
+In order to apply the patch, run
+
+```
+bash tools/patch_linkml
+```
```

Comment:
`hatch` is patching automatically on post-install. This should likely be reworded to an info item and mention that it happens.